aten/src/ATen/native/SoftMax.cpp

```diff
@@ -156,10 +156,7 @@ void host_softmax(
               input_data_base + outer_idx * outer_stride + inner_idx;
           scalar_t* output_data =
               output_data_base + outer_idx * outer_stride + inner_idx;
-          bool* mask_data = nullptr;
-          if (MaskedSoftMax) {
-            mask_data = mask_data_base + outer_idx * outer_stride + inner_idx;
-          }
+          bool* mask_data = mask_data_base + outer_idx * outer_stride + inner_idx;
```

Comment:
can you please investigate why this was causing segfaults? `mask_data` is accessed only if `MaskedSoftMax` is `true`. What you have here potentially has null pointer arithmetics which is an error on llvm 12 (if softmax is not masked, `mask_data_base` should be `nullptr`, right?

Reply:
Looked into this more and I think you're right - I fixed this in the newest commit and I'm not running into segfaults anymore